composer/profiler/__init__.py

```diff
@@ -1,5 +1,52 @@
 # Copyright 2021 MosaicML. All Rights Reserved.
 
-from composer.profiler.event_handler import ProfilerEventHandler as ProfilerEventHandler
-from composer.profiler.profiler import Profiler as Profiler
-from composer.profiler.profiler_action import ProfilerAction as ProfilerAction
+"""The profiler gathers performance metrics during a training run that can be used to diagnose bottlenecks and
+facilitate model development.
+
+The metrics gathered include:
+
+* Duration of each :class:`.Event` during training
+* Time taken by the data loader to return a batch
+* Host metrics such as CPU, system memory, disk and network utilization over time
+* Execution order, latency and attributes of PyTorch operators and GPU kernels (see :doc:`profiler`)
+
+The following example demonstrates how to setup and perform profiling on a simple training run.
+
+.. literalinclude:: ../../../examples/profiler_demo.py
+    :language: python
+    :linenos:
+    :emphasize-lines: 6, 27-49
+
+It is required to specify an output ``profiler_trace_file`` during :class:`.Trainer` initialization to enable profiling.
+The ``profiler_trace_file`` will contain the profiling trace data once the profiling run completes.  By default, the :class:`.Profiler`,
+:class:`.DataloaderProfiler` and :class:`.SystemProfiler` will be active.  The :class:`.TorchProfiler` is **disabled** by default.
+
+To activate the :class:`.TorchProfiler`, the ``torch_profiler_trace_dir`` must be specified *in addition* to the ``profiler_trace_file`` argument.
+The ``torch_profiler_trace_dir`` will contain the Torch Profiler traces once the profiling run completes.  The :class:`.Profiler` will
+automatically merge the Torch traces in the ``torch_profiler_trace_dir`` into the ``profiler_trace_file``, allowing users to view a unified trace.
+
+The complete traces can be viewed by in a Google Chrome browser navigating to ``chrome://tracing`` and loading the ``profiler_trace_file``.
```

Comment:
Can chrome://tracing be in link format?

Reply:
No I originally tried this but it doesn't actually open the trace viewer via a hyperlink.  Not sure if there's some magic to enable which is why I explicitly state how to open the trace viewer in all relevant areas.